### ISSUE TITLE
Not able to downgrade from 1.19.0 > 1.18 or 1.17.

### ISSUE THREAD

**Issue by oakh1** [F5Networks/f5-telemetry-streaming#128]
### Environment
 * Telemetry Streaming Version: 1.19.0
 * BIG-IP Version: 15.1.2.1

### Summary
After upgrading to telemetry streaming 1.19.0 we had some issues with data in the logs missing out, so we had to rollback, but after un-installation of version 1.19.0 the older versions doesn't get installed properly. Checking the GUI they are installed, but when checking for the plugin status using the api, it only gives me a 404 all the time. I've tried to reboot the F5 device but that has made no difference, tried installing back and forth and same issue. I can only get "back" to 1.19.0 but there we have issues with the logging itself...

We have a Splunk consumer which I suspect might be causing this, I read something about a compressionType, we haven't declared it but I see when I check the declaration with the 1.19.0 installed that it's there. How do I disable it since it's there as default? I can't find the proper properties in the schema reference. Or should the uninstallation be enough (it isnt)?

Edit: I tried setting compressionType to "none" before reinstalling but it made no difference. Still only 404 after installing a lower version than 1.19.0. This is causing huge issues with our logging so would be great to know what's causing it.

**Comment by dstokesf5** [comment]
I tried going from 1.19.0 to 1.18.0 and found this error in the restnoded log:
```
Tue, 20 Apr 2021 21:34:40 GMT - severe: [telemetry] Startup Failed
Traceback:
Error: Error: [{"keyword":"enum","dataPath":"/schemaVersion","schemaPath":"#/properties/schemaVersion/enum","params":{"allowedValues":["1.18.0","1.17.0","1.16.0","1.15.0","1.14.0","1.13.0","1.12.0","1.11.0","1.10.0","1.9.0","1.8.0","1.7.0","1.6.0","1.5.0","1.4.0","1.3.0","1.2.0","1.1.0","1.0.0","0.9.0"]},"message":"should be equal to one of the allowed values"}]
    at configUtil.validate.catch.err (/var/config/rest/iapps/f5-telemetry/lib/config.js:203:46)
    at tryCatcher (/usr/share/rest/node/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromiseCtx (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:606:10)
    at Async._drainQueue (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:138:12)
    at Async._drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
Tue, 20 Apr 2021 21:34:40 GMT - severe: [RestWorker] [RestWorker] worker failed to load state: undefined:shared/telemetry
```

It looks like the problem is the schemaVersion property in the Telemetry class. It looks like if you do not specify it, Telemetry Streaming will insert the current version as a default. I was able to work around this with the following steps:
1. Switch back to version 1.19.0
2. Submit a declaration that sets `schemaVersion` to `1.18.0`
3. Downgrade the RPM from 1.19.0 to 1.18.0

We are unlikely to resolve this downgrade issue in older versions, but I am working with the developers to see if we can avoid this issue in future versions.

**Comment by dstokesf5** [comment]
It looks like this was at least partially resolved in version 1.19.0. Future versions should be able to downgrade to version 1.19.0 (or newer) without encountering this start up problem.

**Comment by oakh1** [comment]
I had to change the schemaVersion and also remove the Splunk consumer (as the issue with the compressionType also caused an error (saying that the declaration shouldn't have any additional properties).

After removing Splunk & adding schemaVersion to 1.17.0 I was able to downgrade it.

Thank you : )
